// @patternfly/react-catalog-view-extension/dist/umd/components/VerticalTabs/VerticalTabs.test.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["react", "enzyme", "./VerticalTabs", "./VerticalTabsTab"], factory);
  } else if (typeof exports !== "undefined") {
    factory(require("react"), require("enzyme"), require("./VerticalTabs"), require("./VerticalTabsTab"));
  } else {
    var mod = {
      exports: {}
    };
    factory(global.react, global.enzyme, global.VerticalTabs, global.VerticalTabsTab);
    global.undefined = mod.exports;
  }
})(this, function (_react, _enzyme, _VerticalTabs, _VerticalTabsTab) {
  "use strict";

  var _react2 = _interopRequireDefault(_react);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  test('Vertical Tabs renders tabs properly', () => {
    const component = (0, _enzyme.mount)(_react2.default.createElement(_VerticalTabs.VerticalTabs, {
      id: "vertical-tabs",
      className: "test-vertical-tabs"
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "all",
      className: "test-vertical-tabs-tab",
      title: "All"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "one",
      title: "Tab One"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, null, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one",
      id: "one-one",
      title: "Tab One-One"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, {
      activeTab: true
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-one",
      id: "one-one-one",
      title: "Tab One-One-One",
      active: true
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-two",
      id: "one-one-two",
      title: "Tab One-One-Two",
      active: false
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-three",
      id: "one-one-three",
      title: "Tab One-One-Three"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-two",
      id: "one-two",
      title: "Tab One-Two"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-three",
      id: "one-three",
      title: "Tab One-Three"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "two",
      title: "Tab Two",
      className: "test-tab-class"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, null, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-one",
      id: "two-one",
      title: "Tab Two-One"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-two",
      id: "two-two",
      title: "Tab Two-Two"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-three",
      id: "two-three",
      title: "Tab Two-Three"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "three",
      title: "Tab Three"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, null, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "three-one",
      id: "three-one",
      title: "Tab Three-One"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "three-two",
      id: "three-two",
      title: "Tab Three-Two"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "four",
      title: "Tab Four"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "five",
      title: "Tab Five"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "six",
      title: "Tab Six"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "seven",
      title: "Tab Seven"
    })));
    expect(component).toMatchSnapshot();
  });
  test('Vertical Tabs renders restricted tabs properly', () => {
    const component = (0, _enzyme.mount)(_react2.default.createElement(_VerticalTabs.VerticalTabs, {
      id: "vertical-tabs",
      className: "test-vertical-tabs",
      restrictTabs: true
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "all",
      className: "test-vertical-tabs-tab",
      title: "All",
      shown: true
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "one",
      title: "Tab One",
      hasActiveDescendant: true
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, {
      restrictTabs: true
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one",
      id: "one-one",
      title: "Tab One-One"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, {
      restrictTabs: true,
      activeTab: true
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-one",
      id: "one-one-one",
      title: "Tab One-One-One",
      active: true
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-two",
      id: "one-one-two",
      title: "Tab One-One-Two",
      active: false
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-three",
      id: "one-one-three",
      title: "Tab One-One-Three"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-two",
      id: "one-two",
      title: "Tab One-Two"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-three",
      id: "one-three",
      title: "Tab One-Three"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "two",
      title: "Tab Two",
      className: "test-tab-class"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, {
      restrictTabs: true,
      activeTab: true
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-one",
      id: "two-one",
      title: "Tab Two-One",
      active: true
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-two",
      id: "two-two",
      title: "Tab Two-Two",
      active: false
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "one-one-three",
      id: "two-three",
      title: "Tab Two-Three"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "three",
      title: "Tab Three"
    }, _react2.default.createElement(_VerticalTabs.VerticalTabs, {
      restrictTabs: true
    }, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "three-one",
      id: "three-one",
      title: "Tab Three-One"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      key: "three-two",
      id: "three-two",
      title: "Tab Three-Two"
    }))), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "four",
      title: "Tab Four"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "five",
      title: "Tab Five"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "six",
      title: "Tab Six"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "seven",
      title: "Tab Seven"
    })));
    expect(component).toMatchSnapshot();
  });
  test('Vertical Tabs Tab onActivate is called correctly', () => {
    const onActivateMock = jest.fn();
    const component = (0, _enzyme.mount)(_react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "text-click",
      title: "Click Me",
      onActivate: onActivateMock
    }));
    component.find('#text-click > a').simulate('click');
    expect(component).toMatchSnapshot();
    expect(onActivateMock).toBeCalled();
  });
  test('Vertical Tabs Tab wrap styling is set correctly', () => {
    const component = (0, _enzyme.mount)(_react2.default.createElement("div", null, _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "default-wrap",
      title: "Default Wrap"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "word-wrap",
      title: "Word Wrap",
      wrapStyle: "wrap"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "truncate",
      title: "Truncate",
      wrapStyle: "truncate"
    }), _react2.default.createElement(_VerticalTabsTab.VerticalTabsTab, {
      id: "no-wrap",
      title: "No Wrap",
      wrapStyle: "nowrap"
    })));
    expect(component).toMatchSnapshot();
  });
});
