// @patternfly/react-catalog-view-extension/dist/js/components/VerticalTabs/VerticalTabsTab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VerticalTabsTab = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var React = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var VerticalTabsTab = function VerticalTabsTab(_ref) {
  var _ref$children = _ref.children,
      children = _ref$children === void 0 ? null : _ref$children,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? null : _ref$title,
      _ref$wrapStyle = _ref.wrapStyle,
      wrapStyle = _ref$wrapStyle === void 0 ? 'wrap' : _ref$wrapStyle,
      _ref$active = _ref.active,
      active = _ref$active === void 0 ? false : _ref$active,
      _ref$hasActiveDescend = _ref.hasActiveDescendant,
      hasActiveDescendant = _ref$hasActiveDescend === void 0 ? false : _ref$hasActiveDescend,
      _ref$shown = _ref.shown,
      shown = _ref$shown === void 0 ? false : _ref$shown,
      _ref$onActivate = _ref.onActivate,
      onActivate = _ref$onActivate === void 0 ? null : _ref$onActivate,
      props = _objectWithoutProperties(_ref, ["children", "className", "title", "wrapStyle", "active", "hasActiveDescendant", "shown", "onActivate"]);

  var classes = (0, _classnames["default"])('vertical-tabs-pf-tab', {
    active: active,
    'active-descendant': hasActiveDescendant,
    shown: shown
  }, className);
  var linkClasses = (0, _classnames["default"])({
    'no-wrap': wrapStyle === 'nowrap',
    truncate: wrapStyle === 'truncate'
  });

  var handleActivate = function handleActivate(e) {
    e.preventDefault();

    if (onActivate) {
      onActivate();
    }
  };

  return React.createElement("li", _extends({
    className: classes
  }, props), React.createElement("a", {
    className: linkClasses,
    onClick: function onClick(e) {
      return handleActivate(e);
    },
    href: "#"
  }, title), children);
};

exports.VerticalTabsTab = VerticalTabsTab;
VerticalTabsTab.propTypes = {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  title: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),
  wrapStyle: _propTypes["default"].oneOf(['wrap', 'truncate', 'nowrap']),
  active: _propTypes["default"].bool,
  hasActiveDescendant: _propTypes["default"].bool,
  shown: _propTypes["default"].bool,
  onActivate: _propTypes["default"].func
};
